src/main/java/com/it4045/pokemonviewer/enterprise/service/PokemonServiceStub.java

```diff
@@ -17,9 +18,19 @@
      */
     @Override
     public Pokemon fetchByName(String pokemonName) {
-        pokemon.setPokemonName("Charizard");
-        pokemon.setPokemonId(1);
-        return pokemon;
+        try {
+            for (int i = 0; i < pokemonList.size(); i++) {
+                if (pokemon.getPokemonName().equals(pokemonName)) {
+                    return pokemon;
+                }
+            }
+        } catch(NullPointerException ex) {
+            throw new NullPointerException(ex.getMessage());
+        }
+        return null;
+//      pokemon.setPokemonName("Charizard");
+//      pokemon.setPokemonId(1);
+//      return pokemon;
```

Comment:
Search your pokemonList for a pokemon with a matching name
No need to setPokemonName in fetchByName. Removed those lines.